packages/kit/src/exports/vite/index.js

```diff
@@ -634,8 +684,17 @@ function warn_overridden_config(config, resolved_config) {
  * @param {string[]} out used locally to compute the return value
  */
 function find_overridden_config(config, resolved_config, enforced_config, path, out) {
+	if (
+		typeof config === 'undefined' ||
+		config === null ||
+		typeof resolved_config === 'undefined' ||
+		resolved_config === null
+	) {
```

Comment:
can you explain this logic change?

aside from the bigger picture, not sure why we're doing `typeof` checks here. can we just do this?

```suggestion
	if (config == null || resolved_config == null) {
```

Reply:
We now have two plugins with a `config` method which each provide a partial config, so some of the config values may be `undefined` as they're defined in the other plugin

Reply:
We can probably still do `== null` for each though. The only time I think you'd need to do `typeof foo === 'undefined'` is if `foo` may or may not be a global variable and you were trying to avoid throwing an exception.

Reply:
ok, yeah, that's a good change

man, JS is weird. I guess I failed that question on the JS quiz. I thought that would match `''` or `0` if it matches both `null` and `undefined`, but I guess now